src/Microsoft.Health.Dicom.Functions/BlobMigration/DeleteDurableFunction.Orchestration.cs

```diff
@@ -0,0 +1,82 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Dicom.Core.Features.Model;
+using Microsoft.Health.Dicom.Core.Models.BlobMigration;
+using Microsoft.Health.Dicom.Functions.Indexing.Models;
+using Microsoft.Health.Operations.Functions.DurableTask;
+
+namespace Microsoft.Health.Dicom.Functions.BlobMigration;
+
+public partial class DeleteDurableFunction
+{
+    /// <summary>
+    /// Asynchronously copy DICOM instances.
+    /// It goes through DICOM instances in the past, copy to files as new blob format.
+    /// </summary>
+    /// <param name="context">The context for the orchestration instance.</param>
+    /// <param name="logger">A diagnostic logger.</param>
+    /// <returns>A task representing the <see cref="DeleteFilesAsync"/> operation.</returns>
+    [FunctionName(nameof(DeleteFilesAsync))]
+    public async Task DeleteFilesAsync(
+        [OrchestrationTrigger] IDurableOrchestrationContext context,
+        ILogger logger)
+    {
+        // The ID should be a GUID as generated by the trigger, but we'll assert here just to make sure!
+        EnsureArg.IsNotNull(context, nameof(context)).ThrowIfInvalidOperationId();
+        BlobMigrationCheckpoint input = context.GetInput<BlobMigrationCheckpoint>();
+        EnsureArg.IsNotNull(input, nameof(input));
+        EnsureArg.IsNotNull(input.Batching, nameof(input.Batching));
+
+        logger = context.CreateReplaySafeLogger(logger);
+
+        IReadOnlyList<WatermarkRange> batches = await context.CallActivityWithRetryAsync<IReadOnlyList<WatermarkRange>>(
+            nameof(GetDeleteInstanceBatchesAsync),
+            _options.RetryOptions,
+            new BatchCreationArguments(input.Completed?.Start - 1, input.Batching.Size, input.Batching.MaxParallelCount));
+
+        if (batches.Count > 0)
+        {
+            // Note that batches are in reverse order because we start from the highest watermark
+            // Example: [8,11][4,7][1,3]
+            var batchRange = new WatermarkRange(batches[^1].Start, batches[0].End);
+
+            logger.LogInformation("Beginning to delete the range {Range}.", batchRange);
+            await Task.WhenAll(batches
+                .Select(x => context.CallActivityWithRetryAsync(
```

Comment:
nit: this can probably all go on one line

Reply:
Done